src/Formats/JSONExtractTree.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+#include <DataTypes/IDataType.h>
+#include <Columns/IColumn.h>
+#include <Formats/FormatSettings.h>
+
+
+namespace DB
+{
+
+struct JSONExtractInsertSettings
+{
+    bool convert_bool_to_integer = true;
+    bool insert_default_on_invalid_elements_in_complex_types = false;
```

Comment:
Could you please provide a brief description for why are these new settings are needed? Do we want to handle these differently in json functions and in the new JSON type?

Reply:
These settings are needed to keep the current behaviour of JSONExtract functions.

`convert_bool_to_integer` is set to `false` for functions `JSONExtractUInt64/JSONExtractInt64/JSONExtractFloat64` so we extract only real numbers (don't know the motivation, but we need to keep it as it was).

`insert_default_on_invalid_elements_in_complex_types` is set to true in all `JSONExtract` functions. It allows to do this:
```
:) select JSONExtract('{"a" : [1, "hello", 2]}', 'a', 'Array(Int32)')

SELECT JSONExtract('{"a" : [1, "hello", 2]}', 'a', 'Array(Int32)')

Query id: c401e109-634d-4495-a1bf-831029fac874

   ┌─JSONExtract('{"a" : [1, "hello", 2]}', 'a', 'Array(Int32)')─┐
1. │ [1,0,2]                                                     │
   └─────────────────────────────────────────────────────────────┘
```

So when we can parse only some values from an array/map, we keep them and insert default on the invalid values. Again, it how it was implemented in `JSONExtract`, we had to keep it. But we don't need it in new JSON type